bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,357 @@
+                                    .ofInstant(entry.getEnd().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
+                            break;
+
+                    }
+                    updateState(channelUID, result);
+
+                } catch (SVDRPParseResponseException e) {
+                    logger.trace("VDR Refresh for Thing {}, ChannelUID {}, Parse Response failed. Message: {}",
+                            this.getThing().getUID(), channelUID, e.getMessage());
+                    updateState(channelUID, UnDefType.UNDEF);
+                } catch (SVDRPConnectionException e) {
+                    logger.debug("VDR Refresh for Thing {}, ChannelUID {}, Connection failed. Message: {}",
+                            this.getThing().getUID(), channelUID, e.getMessage());
+                }
+
+            });
+
+        } catch (SVDRPException ce) {
+            if (thing.getStatus() == ThingStatus.ONLINE) {
+                // also update power channel when thing is offline before setting it offline
+                thing.getChannels().stream().map(c -> c.getUID()).filter(this::isLinked).forEach(channelUID -> {
+                    try {
+                        if ("power".equals(channelUID.getIdWithoutGroup())) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } catch (Exception e) {
+                        logger.warn("VDR Refresh for Thing {}, ChannelUID {} failed. ErrorMessage: {}",
+                                this.getThing().getUID(), channelUID, e.getMessage());
+                    }
+                });
+            }
+            updateStatus(ThingStatus.OFFLINE);
+
+        } catch (Exception ex) {
+            logger.error("There is something wrong with your thing '{}', please check or recreate it: {}",
+                    thing.getUID(), ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ex.getMessage());
+        } finally {
+            try {
+                con.closeConnection();
+            } catch (SVDRPException e) {
+                logger.trace("Error on VDR Refresh while closing SVDRP Connection for Thing : {} with message {}",
+                        this.getThing().getUID(), e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Schedules the refresh thread
+     */
+    private void scheduleRefreshThread() {
+        refreshThreadFuture = scheduler.scheduleWithFixedDelay(this::onVDRRefresh, 3, config.getRefresh(),
+                TimeUnit.SECONDS);
+    }
+
+    /**
+     * Stops the refresh thread.
+     *
+     * @param force if set to true thread cancellation will be forced
+     */
+    @SuppressWarnings({ "null" })
+    private void stopRefreshThread(boolean force) {
+        if (refreshThreadFuture != null) {
+            refreshThreadFuture.cancel(force);
+        }
```

Comment:
You could store `refreshThreadFuture` to a local variable an do the null check on that. Then, you can remove the null supression.

Reply:
Can you please have a look at my change if I understood correctly?